pkg/install/v1alpha1/installer.go

```diff
@@ -140,6 +149,48 @@ func (i *simpleInstaller) Install() []error {
 	return i.errors
 }
 
+// cleanResources are the versioned openebs respurces that needs to cleaned
+// after installing the latest versions of the respective resource
+var cleanResources = []resourceMetadata{
+	resourceMetadata{
+		name: "castemplates",
+	},
+	resourceMetadata{
+		name:      "runtasks",
+		namespace: menv.Get(menv.OpenEBSNamespace),
+	},
+}
+
+// Clean is the cleanup function that removes old version of openebs resources
+func (i *simpleInstaller) Clean() error {
+	for _, res := range cleanResources {
+		gvr := schema.GroupVersionResource{
+			Group:    "openebs.io",
+			Version:  "v1alpha1",
+			Resource: res.name,
+		}
+		// the label selector prevents clean up of resources on each restart
+		// as the listed items will be an empty list.
+		resList, err := k8s.ListResource(gvr, res.namespace).
+			List(metav1.ListOptions{
+				LabelSelector: "version!=" + version.Current(),
+			})
+		if err != nil {
+			return err
+		}
+		for _, obj := range resList.Items {
```

Comment:
Instead fetching and looping through resources, we can use api from `pkg/client/k8s/v1alpha1` to delete runtask.
```
k8sDynamic, err := k8s.Dynamic.Provide()
// err handling
err = k8sDynamic.Resource(gvr).Namespace(res.namespace).DeleteCollection(
       metav1.DeleteOptions{},
       metav1.ListOptions{
              LabelSelector: "version!=" + version.Current(),
       },
)
// err handing
```
```

Reply:
Done